Source/Core/CesiumTerrainProvider.js

```diff
@@ -489,17 +493,16 @@ define([
      * @param {Number} x The X coordinate of the tile for which to request geometry.
      * @param {Number} y The Y coordinate of the tile for which to request geometry.
      * @param {Number} level The level of the tile for which to request geometry.
-     * @param {Boolean} [throttleRequests=true] True if the number of simultaneous requests should be limited,
-     *                  or false if the request should be initiated regardless of the number of requests
-     *                  already in progress.
+     * @param {Request} [request] The request object.
+     *
      * @returns {Promise.<TerrainData>|undefined} A promise for the requested geometry.  If this method
      *          returns undefined instead of a promise, it is an indication that too many requests are already
      *          pending and the request will be retried later.
      *
      * @exception {DeveloperError} This function must not be called before {@link CesiumTerrainProvider#ready}
      *            returns true.
      */
-    CesiumTerrainProvider.prototype.requestTileGeometry = function(x, y, level, throttleRequests) {
+    CesiumTerrainProvider.prototype.requestTileGeometry = function(x, y, level, request) {
```

Comment:
Right now every terrain/imagery provider sets `throttleByServer` to true which limits to 6 active requests per server. Once more of these servers switch over to http2 this flag will hold them back. With `CesiumTerrainProvider` and some others we'll know when the make the switch, but this doesn't help people using custom providers. Should an option like `throttleByServer` be exposed on the provider level for users to tweak?

This same idea applies to 3D tilesets too. A `throttleByServer` option could be sent in when constructing the tileset.